stack pause feature

annie-xd-wang opened the pull request:
#398, add an independent stack pause feature and add it to the current ZStackAcquisition workflow.

annie-xd-wang committed:
add StackPause feature

annie-xd-wang committed:
add StackPause to ZStackAcquisition

annie-xd-wang committed:
experiment time includes stack pauses

AdvancedImagingUTSW committed:
Update channels_tab_controller.py

Co-Authored-By: Annie Wang <6161065+annie-xd-wang@users.noreply.github.com>

AdvancedImagingUTSW committed:
add stackpause to confocal projection mode

Co-Authored-By: Annie Wang <6161065+annie-xd-wang@users.noreply.github.com>

zacsimile reviewed (approved):
Looks good and confirmed working on the multiscale.

AdvancedImagingUTSW committed:
Update daq_ni.py

Looking for conflicts...  Had to do some formatting.

AdvancedImagingUTSW committed:
Updating from develop

Merged

annie-xd-wang committed:
revert commit: add stackpause to confocal projection mode

Confocal projection with stackpause will be added later because of speed concern. Another branch "opmv2-zstack-stackpause" supports confocal projection with stackpause now.

annie-xd-wang committed:
small fix

annie-xd-wang committed:
no pause after the last stack

annie-xd-wang committed:
frequently check whether to stop acquisition in a stack pause

annie-xd-wang committed:
check whether to stop pause every 100ms

It still need time to resume data thread before it stops completely.

AdvancedImagingUTSW committed:
Merge branch 'develop' into 398-stack-pause-feature
